lib/AST/ASTContext.cpp

```diff
@@ -3098,7 +3098,7 @@ AnyFunctionType::Param swift::computeSelfParam(AbstractFunctionDecl *AFD,
 
   if (auto *FD = dyn_cast<FuncDecl>(AFD)) {
     isStatic = FD->isStatic();
-    isMutating = FD->isMutating();
+    isMutating = FD->isMutating() && !containerTy->hasReferenceSemantics();
```

Comment:
This is unnecessary because isMutating() will never be true for a class method.

Reply:
I’ll change it to an assertion.